### DISCUSSION TITLE
How to customize Palette, increase number of color and color square size?

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#531] [General]
Assume I load a *.jpg or *.png image into PhotoDemon and want to use some (new or existing) colors to fill an area or add some colored text. 

For that purpose I prefer mostly not to use the full color wheel.
Instead I would like to have a pre-defined customized color palette at the side pane (see blue arrow in attached snapshot).

Currently I can see the only 14 available colors. 

How can I increase the number of shown colors from 14 to lets say 32 or 64 colors?

How can I increase the square sizes to lets say the doubled size?

How can I change to colors in this pre-defined palette?

How can I export and import this customized palette and make it my default?

For clarification: My favorite colors should be independent from the colors in the current image, no matter if is a 256 color, a 8bit color depth or 16 bit color depth RGB image.

How can I achieve this?

![PalettePD](https://github.com/tannerhelland/PhotoDemon/assets/2775875/9a2ba7ce-208e-444f-888d-c1c3019e70b4)

**Comment by hi5** [comment]
I see two options, you create your own image with all the colours you want, just pixels / squares. Then create a palette from image.

Second option is to create one manually from hex html color codes:

If you have already created a PD palette file you can find these in `Data\Palettes\`
You can open these in notepad or another text editor. I don't know the max but below you'll find one with 139 colors which I created in notepad.

The first six chars are the hex html codes of the colour, followed by `ff` - which I assume is an alpha setting or something (_wild guess_)

![htmltable](https://github.com/tannerhelland/PhotoDemon/assets/1973899/dc177660-f1be-4f92-af09-501af6f5e267)

htmltable.pdpalette: [htmltable.zip](https://github.com/tannerhelland/PhotoDemon/files/14151803/htmltable.zip)

**Reply by pstein** [reply]
Perfect!
Thank you

Is it possible to show the palette and the color wheel at the same time?
In case that I need an additional color  I always have to switch (temporarily) manually to color wheel and back.

**Comment by hi5** [comment]
As far as I know it is either the palette _or_ the colour wheel, but I can't really think of a reason - other than being "because it is so" - it couldn't be changed to:

colour wheel
history recent 16 [where I would suggest to change it to 2x8 squares so they are bigger, reason being with higher screen resolutions these do become quite tiny)
palette, and the palette is something you can turn on/off

Something like:
![new-colour-setup](https://github.com/tannerhelland/PhotoDemon/assets/1973899/efc8692f-a2ad-46ec-8359-acc9a62ba4cd)

So always colour wheel + history and optional palette.

**Reply by pstein** [reply]
Thank you
